src/zcash/circuit/utils.tcc

```diff
@@ -0,0 +1,55 @@
+template<typename FieldT>
+pb_variable_array<FieldT> from_bits(std::vector<bool> bits, pb_variable<FieldT>& ZERO) {
+    pb_variable_array<FieldT> acc;
+
+    BOOST_FOREACH(bool bit, bits) {
+        acc.emplace_back(bit ? ONE : ZERO);
+    }
+
+    return acc;
+}
+
+std::vector<bool> trailing252(std::vector<bool> input) {
+    if (input.size() != 256) {
+        throw std::length_error("trailing252 input invalid length");
+    }
+
+    return std::vector<bool>(input.begin() + 4, input.end());
+}
+
+std::vector<bool> uint256_to_bool_vector(uint256 input) {
+    std::vector<unsigned char> input_v(input.begin(), input.end());
+    std::vector<bool> output_bv(256, 0);
+    libzerocash::convertBytesVectorToVector(
+        input_v,
+        output_bv
+    );
+
+    return output_bv;
+}
+
+std::vector<bool> uint64_to_bool_vector(uint64_t input) {
+    std::vector<unsigned char> num_bv(8, 0);
+    libzerocash::convertIntToBytesVector(input, num_bv);
+    std::vector<bool> num_v(64, 0);
+    libzerocash::convertBytesVectorToVector(num_bv, num_v);
```

Comment:
Note that byte->bit conversion is most-significant-bit first, so only change the name of `convertIntToBytesVector` to reflect that it is little-encroach, not the name of `uint64_to_bool_vector` (which is mixed-endian) or `convertBytesVectorToVector`. Although the latter is a terrible name for other reasons (it should say that it's converting to bits).